db/compaction/compaction_job.cc

```diff
@@ -2009,11 +1383,12 @@ Status CompactionJob::FinishCompactionOutputFile(
     // "normal" status, it does not also need to be checked
     sub_compact->io_status.PermitUncheckedError();
   }
-  sub_compact->outfile.reset();
+
+  sub_compact->ResetFileWriter();
```

Comment:
There are lots of GetTableBuilder(), GetFileWriter(), etc in this function. Can most code in this function moved to class CompactionOutputs? We don't necessarily do it all in this PR though but consider to do it as a follow up.

Reply:
All these functions are now moved inside of `CompactionOutputs`.

Ideally `Open/FinishCompactionOutputFile` should be refactored into `CompactionOutputs` too, added TODO for that.